src/components/Login/Web3Login.tsx

```diff
@@ -351,34 +363,58 @@ const Web3Login: FC<Props> = ({
 		onWalletUpdate && onWalletUpdate();
 	};
 
+	const handleChangeWalletWithPolkasafe = (wallet:string) => {
+		setChosenWallet(wallet);
+		console.log(wallet);
+		setAccounts([]);
+	};
+	useEffect(() => {
+		if(withPolkasafe && accounts.length === 0 && chosenWallet !== Wallet.POLKASAFE){
+			getAccounts(chosenWallet)
+				.then(() => {setLoading(false);setFetchAccounts(false);})
+				.catch((err) => {
+					console.error(err);
+				});
+		}
+	// eslint-disable-next-line react-hooks/exhaustive-deps
+	},[accounts.length, chosenWallet, withPolkasafe]);
+
 	return (
 		<>
-			<article className="bg-white shadow-md rounded-md flex flex-col gap-y-3">
-				<div className='flex items-center px-4 pt-4 pb-2'>
-					<LoginLogo className='mr-3' />
-					<span className="text-[20px] font-semibold text-bodyBlue">Login</span>
-				</div>
-				<hr className='border-[#D2D8E0] border-opacity-30 -mt-2' />
-				<h3 className="text-2xl font-semibold text-[#1E232C] flex flex-col px-4 mb-0">
-					{/* <span>Login</span> */}
-					<p className='flex gap-x-2 items-center justify-start p-0 m-0'>
-						<span>
+			<div className='flex items-center'>
+				<LoginLogo className='ml-6 mr-2' />
+				<h3 className="text-[20px] font-semibold text-[#243A57] mt-3">{withPolkasafe ? <PolkasafeWithIcon/> : 'Login'}</h3>
+			</div>
+			<hr className='text-[#D2D8E0] ' />
+			<article className="bg-white shadow-md rounded-md p-8 flex flex-col gap-y-3">
+				<h3 className="text-2xl font-semibold text-[#1E232C] flex flex-col gap-y-2">
+					{!withPolkasafe &&<p className='flex gap-x-2 items-center justify-start p-0 m-0'>
+						<span className='mt-2'>
 							<WalletIcon which={chosenWallet} />
 						</span>
-						<span className='text-bodyBlue text-lg sm:text-xl'>
+						<span className='text-[#243A57] text-lg sm:text-xl'>
```

Comment:
arbitrary colour, please try and use design standards.
@sahas-01 might be able to help.